pkg/storaged/dialog.jsx

```diff
@@ -998,129 +1021,229 @@ export const SizeSlider = (tag, title, options) => {
+                          ]}
+                          rows={rows} />
+        </div>);
+};
 
-    add_usage_message(parts, teardown.MDRaidMembers,
-                      _("This device is currently used for RAID devices. Proceeding will remove it from its RAID devices."),
-                      "Name", "MDRaid");
+export function init_active_usage_processes(client, usage, top_action, child_action) {
+    return {
+        title: _("Checking related processes"),
+        func: dlg => {
+            return for_each_async(usage, u => {
+                if (u.usage == "mounted") {
+                    return client.nfs.entry_users({
+                        fields: [null, decode_filename(u.fsys.MountPoints[0])],
+                        mounted: true
+                    })
+                            .then(users => {
+                                u.users = users;
+                            });
+                } else
+                    return Promise.resolve();
+            }).then(() => {
+                dlg.set_attribute("Teardown", TeardownMessage(usage, top_action, child_action));
+                const usage_with_users = usage.filter(u => u.users);
+                const n_processes = usage_with_users.reduce((sum, u) => sum + u.users.filter(u => u.pid).length, 0);
+                const n_services = usage_with_users.reduce((sum, u) => sum + u.users.filter(u => u.unit).length, 0);
+                if (n_processes > 0 && n_services > 0)
+                    dlg.add_danger(_("Related processes and services will be forcefully stopped."));
+                else if (n_processes > 0)
+                    dlg.add_danger(_("Related processes will be forcefully stopped."));
+                else if (n_services > 0)
+                    dlg.add_danger(_("Related services will be forcefully stopped."));
+            });
+        }
+    };
+}
 
-    const has_sessions = teardown.Sessions && teardown.Sessions.length > 0;
-    const has_services = teardown.Services && teardown.Services.length > 0;
+export const StopProcessesMessage = ({ mount_point, users }) => {
+    const process_rows = users.filter(u => u.pid).map(u => {
+        return {
+            columns: [
+                u.pid,
+                u.cmd.substr(0, 100),
+                u.user || "-",
+                format_delay(-u.since * 1000)
+            ]
+        };
+    });
 
-    if (has_sessions && has_services)
-        add_para(parts, _("The filesystem is in use by login sessions and system services. Proceeding will stop these."));
-    else if (has_sessions)
-        add_para(parts, _("The filesystem is in use by login sessions. Proceeding will stop these."));
-    else if (has_services)
-        add_para(parts, _("The filesystem is in use by system services. Proceeding will stop these."));
+    const service_rows = users.filter(u => u.unit).map(u => {
+        return {
+            columns: [
+                u.unit.replace(/\.service$/, "").substr(0, 40),
```

Comment:
It's plausible to have services longer than 40 characters. Would it be possible to limit these via CSS instead? max-width and `text-overflow: ellipsis` or so? (and thus showing the full name in a tooltip)

Not a blocker though, fine to do this as a follow-up.

Reply:
I give it a try...

Reply:
Oh, Patternfly has nice "modifiers" for Table cells.  `modifier="truncate"` would show ellipses, but let's use "breakWord" so that nothing gets hidden but long words will be broken in their middle.  (And let's use "nowrap" for the "Runtime" column.)

Super long command line (one of Cockpits embedded python scripts that we now load via the command line):

![image](https://user-images.githubusercontent.com/3228183/148578659-7a6ad0df-401d-42e5-bc0e-8f77d810a909.png)

Super long service name and description (cockpits default https instance):

![image](https://user-images.githubusercontent.com/3228183/148578891-6d07e374-e7a7-459f-be96-9c9bca39baab.png)

When we force the table columns to be 25/25/25/25 percent:

![image](https://user-images.githubusercontent.com/3228183/148579104-7d1400b9-0884-49d6-a4a8-74a6f01f7786.png)
![image](https://user-images.githubusercontent.com/3228183/148579148-0cb88c7b-c08b-4f53-8a13-fbeb296aaa45.png)

Reply:
Fixed.

Reply:
Superb, thanks!